MFVNeutralino/plugins/EventHistos.cc

```diff
@@ -309,9 +343,85 @@ void MFVEventHistos::analyze(const edm::Event& event, const edm::EventSetup&) {
       h_bquark_pairdphi->Fill(reco::deltaPhi(mevent->gen_bquarks[i].Phi(), mevent->gen_bquarks[j].Phi()), w);
   }
 
+  for (int igenv = 0; igenv < 2; ++igenv) {
+    double genx = mevent->gen_lsp_decay[igenv*3+0];
+    double geny = mevent->gen_lsp_decay[igenv*3+1];
+    double genz = mevent->gen_lsp_decay[igenv*3+2];
+    double genbs2ddist = mevent->mag(genx - mevent->bsx_at_z(genz),
+                                     geny - mevent->bsy_at_z(genz) 
+        );
+    h_gen_bs2ddist->Fill(genbs2ddist, w);
+  }
+
   h_minlspdist2d->Fill(mevent->minlspdist2d(), w);
   h_lspdist2d->Fill(mevent->lspdist2d(), w);
   h_lspdist3d->Fill(mevent->lspdist3d(), w);
+  h_llp_dphi->Fill(mevent->gen_lsp_phi[0]-mevent->gen_lsp_phi[1], w);
+
+  h_llp_pt_vecsum->Fill((mevent->gen_lsp_p4(0)+mevent->gen_lsp_p4(1)).Pt(), w);
+  h_llp0pt_llp1pt->Fill(std::max(mevent->gen_lsp_pt[0], mevent->gen_lsp_pt[1]), std::min(mevent->gen_lsp_pt[0], mevent->gen_lsp_pt[1]), w);
+  double decay_quarks0_pt[2] = {-1,-1};
+  double decay_quarks1_pt[2] = {-1,-1};
+  double decaylsp0_pt = -1;
+  double decaylsp1_pt = -1;
+  double decay_jets0_pt[2] = {-1,-1};
+  double decay_jets1_pt[2] = {-1,-1};
+  double nmatched_0 = 0;
+  double nmatched_1 = 0;
+  for (size_t i=0; i<mevent->gen_daughters.size(); ++i){
+    if (abs(mevent->gen_daughter_id[i])==1000022){
+      if (decaylsp0_pt>0){
+        decaylsp1_pt = mevent->gen_daughters[i].Pt();
+      }
+      else{
+        decaylsp0_pt = mevent->gen_daughters[i].Pt();
+      }
+    }
+    else{
+      double gd_eta = mevent->gen_daughters[i].Eta();
+      double gd_phi = mevent->gen_daughters[i].Phi();
+      //double dR2_min = 999;
+      int n_matched = 0;
+      double pt_sum = 0;
+      for (int ij = 0; ij<MAX_NJETS; ++ij){
+        double dR2 = (mevent->nth_jet_eta(ij)-gd_eta)*(mevent->nth_jet_eta(ij)-gd_eta)+(mevent->nth_jet_phi(ij)-gd_phi)*(mevent->nth_jet_phi(ij)-gd_phi);
+        //if (dR2_min>dR2){
+        //  dR2_min = dR2;
+        //}
+        if (dR2<0.16){
+          n_matched += 1;
+          pt_sum += mevent->nth_jet_pt(ij);
+        }
+      }
```

Comment:
I don't remember why that pdgID is handled differently than the others, so comments might be useful here

Reply:
I think the reason is 1000022 is neutralino, which is not daughters in other benchmark samples. I will add comments on that.

Reply:
Oh good catch! I was thinking about the neutralino in tbs multijet samples, but it's the primary particle rather than a daughter...